src/coreclr/jit/lower.cpp

```diff
@@ -1842,6 +1842,155 @@ GenTree* Lowering::AddrGen(void* addr)
     return AddrGen((ssize_t)addr);
 }
 
+// LowerCallMemset: Replaces the following memset-like special intrinsics:
+//
+//    SpanHelpers.Fill<T>(ref dstRef, CNS_SIZE, 0)
+//    SpanHelpers.ClearWithoutReferences(ref dstRef, CNS_SIZE)
+//
+//  with a GT_STORE_BLK node:
+//
+//    *  STORE_BLK struct<CNS_SIZE> (init) (Unroll)
+//    +--*  LCL_VAR   byref  dstRef
+//    \--*  CNS_INT   int    0
+//
+// Arguments:
+//    tree - GenTreeCall node to replace with STORE_BLK
+//    next - [out] Next node to lower if this function returns true
+//
+// Return Value:
+//    false if no changes were made
+//
+bool Lowering::LowerCallMemset(GenTreeCall* call, GenTree** next)
+{
+    JITDUMP("Considering Memset-like call [%06d] for unrolling.. ", comp->dspTreeID(call))
+
+    if (comp->info.compHasNextCallRetAddr)
+    {
+        JITDUMP("compHasNextCallRetAddr=true so we won't be able to remove the call - bail out.\n");
+        return false;
+    }
+
+    // void SpanHelpers::Fill<T>(ref T dstRef, nuint numElements, T value)
+    // void SpanHelpers::ClearWithoutReferences(ref byte dstRef, nuint byteLength)
+
+    GenTree* dstRefArg = call->gtArgs.GetUserArgByIndex(0)->GetNode();
+    GenTree* lengthArg = call->gtArgs.GetUserArgByIndex(1)->GetNode();
+    GenTree* valueArg  = nullptr;
+
+    if (!lengthArg->IsIntegralConst())
+    {
+        JITDUMP("Length is not a constant - bail out.\n")
+        return false;
+    }
+
+    // Fill<T>'s length is not in bytes, so we need to scale it depending on the signature
+    unsigned lengthScale;
+
+    const NamedIntrinsic ni = comp->lookupNamedIntrinsic(call->gtCallMethHnd);
+    if (ni == NI_System_SpanHelpers_Fill)
+    {
+        // void SpanHelpers::Fill<T>(ref T refData, nuint numElements, T value)
+        //
+        assert(call->gtArgs.CountUserArgs() == 3);
+        valueArg = call->gtArgs.GetUserArgByIndex(2)->GetNode();
+
+        // Get that <T> from the signature
+        CORINFO_SIG_INFO sig;
+        comp->info.compCompHnd->getMethodSig(call->gtCallMethHnd, &sig, nullptr);
+        assert(sig.sigInst.methInstCount == 1);
+        lengthScale = genTypeSize(
+            JITtype2varType(comp->info.compCompHnd->getTypeForPrimitiveValueClass(sig.sigInst.methInst[0])));
```

Comment:
This can just use `GetSignatureType()` from the `CallArg` you already fetch to get the value node.